### PR TITLE
feat: [CDS-42111]: updating ticket fields to approval instance on every polling event

### PR DESCRIPTION
**Pull request by naman-goenka**
## Describe your changes
- updated ticket fields to approval instance, currently only status fields as decided by product.
- can be extended to other fields in future (providing entire approval instance including criteria to update method).
- only update when we have status field in ticket and when updated value is not present in db.
- warning console logs when exception occurs while update. 
- this fields will be included in getApproval call, which UI has to poll to fetch updated fields. 

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44596)

### CODE REVIEW

**Review by puthrayaharness — APPROVED** [review]

File: pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/approval/step/ApprovalInstanceResponseMapper.java [hunk 1]
```diff
@@ -118,10 +121,18 @@ private ApprovalInstanceDetailsDTO toServiceNowApprovalInstanceDetailsDTO(Servic
         AmbianceUtils.getAccountId(instance.getAmbiance()), AmbianceUtils.getOrgIdentifier(instance.getAmbiance()),
         AmbianceUtils.getProjectIdentifier(instance.getAmbiance()), instance.getConnectorRef());
 
+    Map<String, String> fields;
+    if (!isEmpty(instance.getTicketFields())) {
+      fields = new HashMap<>();
+      instance.getTicketFields().forEach((k, v) -> fields.put(k, v.getDisplayValue()));
+    } else {
+      fields = null;
```
Comment (sribalij) [inline pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/approval/step/ApprovalInstanceResponseMapper.java:129]: Can remove `else`?

**Comment by naman-goenka** [comment]
> Add some Junits?

Added junits. was waiting for approach validation.

File: pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/approval/step/ApprovalInstanceResponseMapper.java (hunk 1 above)
Comment (naman-goenka) [reply pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/approval/step/ApprovalInstanceResponseMapper.java:129]: if we try to assign null at line 124, then the lambda expression on line 127 complains that fields variable should be final (in lambda expressions). So, added the initialisation here, to not depend on default compiler initialisation for local variables. 


Can you please point if there is a better approach possible here.

**Review by puthrayaharness — APPROVED** [review]

**Review by puthrayaharness — DISMISSED** [review]
> Reviewed 10 of 11 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44596)* status: 11 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by shivamnegi94 — APPROVED** [review]

**Review by shivamnegi94 — APPROVED** [review]
> Reviewed 10 of 11 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44596)* status: Some of the checks are too old. 1. unresolved discussions.

**Review by vtxorxwitty — APPROVED** [review]
> Reviewed 10 of 11 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44596)* status: 1. unresolved discussions.

**Pull request merged**